### PR TITLE
Make sqlcipher an optional dependency for everything except android logins library. Fixes #234

### LINKED ISSUE #234: Allow logins-sql to be built without requiring sqlcipher.
Follow up for #221 because we don't need it for the milestone.

This is trickier than expected to make it work without it silently ignoring your requests for sqlcipher/non-sqlcipher without letting you know.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-715)
┆Epic: a-s storage components (backlog)
┆Sprint: Backlog

### PR DESCRIPTION
**Pull request by thomcc**

### CODE REVIEW

**Comment by thomcc** [comment]
The failure there doesn't list any errors, and says everything passed, which is... :|

That said, this causes weird things to happen when no features are specified, so it might need more thought.

**Comment by ncalexan** [comment]
> The failure there doesn't list any errors, and says everything passed, which is... :|

I think this is just a shell weirdness, or a Travis weirdness: those commands aren't chained with `&&` or `set -e`.

And
```
thread 'main' panicked at 'Builds with bundled SQLCipher are not supported', /Users/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/libsqlite3-sys-0.9.3/build.rs:13:13
```
is pretty clear.  (That's from the Nightly Rust build.)  Was this ever supposed to work?

In any case, I'm pleased that this isn't failing with the changes to the `rust-android-gradle` plugin.

File: places/Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,11 @@ name = "places"
 version = "0.1.0"
 authors = []
 
+[features]
+bundled = ["rusqlite/bundled"]
```
Comment (ncalexan) [inline places/Cargo.toml]: Ah, I see -- this probably explains the Travis error.  Can you investigate?

**Comment by thomcc** [comment]
Ah, yeah I see, we probably need a `cargo clean` between the builds that use sqlcipher and the ones that don't.

File: places/Cargo.toml (hunk 1 above)
Comment (thomcc) [reply places/Cargo.toml]: Right, we're doing an --all-features in the travis tests. Ugh.

**Comment by thomcc** [comment]
The PR that will fix this will look different from this because it's been too long.

**Pull request closed without merging**